### PR TITLE
Deterministic stdout output. Resolves #8.

### LINKED ISSUE #8: Nondeterminism makes testing expected output hard
Currently the output of bin/decompile can differ each time it is run on the same input. This is likely due to our use of unordered collections such as `dict`. Unfortunately, this makes output testing of the kind being implemented in [lexibrent:bats-tests](https://github.com/lexibrent/vandal/tree/bats-tests) impossible.

The solution I propose is updating the functions that produce textual output so that all non-deterministically ordered data structures are sorted.

For example, recently I saw these two different outputs for `basic_optimized.dasm`:

```
5c5
< Successors: [0x1a, 0x18]
---
> Successors: [0x18, 0x1a]
108a109
> --
```

My proposed solution is to sort the preds and succs lists, and any other relevant structures in our output rendering functions.

### PR DESCRIPTION
**Pull request by 0xl3x1**
* Use `sorted()` for sets in `__str__` functions
* Make BasicBlocks sortable by implementing `__lt__`

Resolves #8

### CODE REVIEW

**Comment by 0xl3x1** [comment]
Upon further testing I just realised there's some issues with function extraction. I'll need to resolve those before this is merged.

**Comment by 0xl3x1** [comment]
@zyzek OK, I think this is ready for review now :+1:

File: src/cfg.py [hunk 1]
```diff
@@ -309,11 +309,19 @@ def __str__(self):
...
+        succ = "Successors: [{}]".format(", ".join(b.ident() for b in sorted(self.succs)))
         unresolved = "\nHas unresolved jump." if self.has_unresolved_jump else ""
         return "\n".join([head, self._STR_SEP, pred, succ]) + unresolved
 
+    def __lt__(self, other):
+        """
+        Compare BasicBlocks based on their entry program counter values.
+        """
+        if self.entry is None or other.entry is None:
+            return True
```
Comment (zyzek) [inline src/cfg.py]: What's the rationale behind returning True in this case?

File: src/dataflow.py [hunk 2]
```diff
@@ -306,11 +306,11 @@ def stack_analysis(cfg: tac_cfg.TACGraph) -> bool:
                                                    for block in cfg.blocks}
                     if settings.clamp_large_stacks:
                         unmod_stack_changed_count = 0
-                        for succ in curr_block.succs:
+                        for succ in sorted(curr_block.succs):
```
Comment (0xl3x1) [inline src/dataflow.py]: Actually, probably don't need this one. I'll go through and remove the overly cautious ones

File: src/cfg.py (hunk 1 above)
Comment (0xl3x1) [reply src/cfg.py]: I guess it could be false. I chose true so that `None` appears at the start of a sorted list.

**Review by zyzek — APPROVED** [review]

**Pull request merged**